bazel/envoy_library.bzl

```diff
@@ -80,13 +80,17 @@ EXTENSION_STATUS_VALUES = [
 def envoy_cc_extension(
         name,
         security_posture,
+        category = None,
         # Only set this for internal, undocumented extensions.
         undocumented = False,
         status = "stable",
         tags = [],
         extra_visibility = [],
         visibility = EXTENSION_CONFIG_VISIBILITY,
         **kwargs):
+    if not category:
+        fail("Category not set for %s" % name)
```

Comment:
@htuch im wondering about matching against a list of known categories, which i think you mentioned

should i use the command i posted to create a list and add this as a constant or somesuch?

Reply:
Yep, adding an explicit list here (i.e. as done for security posture) will make this more robust.

Reply:
cool once i know what the definitive list is (or how best to create it) i will add

Reply:
and add handling for iterables

Reply:
added